src/univariate/continuous/chernoff.jl

```diff
@@ -0,0 +1,318 @@
+        -1.1015045702774968
+         1.1165961779326556
+        -1.1307323104931881
+         1.1440366732735523
+        -1.156609849116565
+        ]
+
+    const cuberoottwo = cbrt(2.0)
+    const sqrthalfpi = sqrt(0.5*pi)
+    const sqrttwopi = sqrt(2.0*pi)
+
+    function p(y::Real)
+        if iszero(y)
+            return -sqrt(0.5*pi) 
+        end
+
+        (y > 0) || throw(DomainError(y, "argument must be positive"))
+
+        cnsty = y^(-1.5)
+        if (y <= 1.0)
+            return sum([(b[k]*cnsty - a[k]*sqrthalfpi)*y^(3*k) for k=1:length(a)])-sqrthalfpi
+        else
+            return sum([exp(cuberoottwo*a*y) for a in airyai_roots]) * 2 * sqrttwopi * exp(-y*y*y/6) - cnsty
+        end
+    end
+
+    function g(x::Real)
+        function g_one(y::Real)
+            return p(y) * exp(-0.5*y*(2*x+y)*(2*x+y))
+        end
+        function g_two(y::Real)
+            z = 2*x+y*y
+            return (z*y*y + 0.5 * z*z) * exp(-0.5*y*y*z*z)
+        end
+        if (x <= -1.0)
+            return cuberoottwo*cuberoottwo * exp(2*x*x*x/3.0) * sum([exp(-cuberoottwo*airyai_roots[k]*x) / airyai_prime[k] for k=1:length(airyai_roots)])
+        else
+            return 2*x - (quadgk(g_one, 0.0, Inf)[1] - 4*quadgk(g_two, 0.0, Inf)[1]) / sqrttwopi   # should perhaps combine integrals
+        end
+    end
+
+    _pdf(x::Real) = g(x)*g(-x)*0.5
+    _cdf(x::Real) = (x < 0.0) ? _cdfbar(-x) : 0.5 + quadgk(_pdf,0.0,x)[1] 
+    _cdfbar(x::Real) = (x < 0.0) ? _cdf(x) : quadgk(_pdf, x, Inf)[1]
+end
+
+pdf(d::Chernoff, x::Real) = ChernoffComputations._pdf(x)
+logpdf(d::Chernoff, x::Real) = log(ChernoffComputations.g(x))+log(ChernoffComputations.g(-x))+log(0.5)
+cdf(d::Chernoff, x::Real) = ChernoffComputations._cdf(x)
+
+function quantile(d::Chernoff, tau::Real)
+    # some commonly used quantiles were precomputed
+    precomputedquants=[
+        0.0 -Inf;
+        0.01 -1.171534341573129;
+        0.025 -0.9981810946684274;
+        0.05 -0.8450811886357725;
+        0.1 -0.6642351964332931;
+        0.2 -0.43982766604886553;
+        0.25 -0.353308035220429;
+        0.3 -0.2751512847290148; 
+        0.4 -0.13319637678583637;
+        0.5 0.0;
+        0.6 0.13319637678583637;
+        0.7 0.2751512847290147;
+        0.75 0.353308035220429;
+        0.8 0.4398276660488655;
+        0.9 0.6642351964332931;
+        0.95 0.8450811886357724;
+        0.975 0.9981810946684272;
+        0.99 1.17153434157313;
+        1.0 Inf
+        ]
+    (0.0 <= tau && tau <= 1.0) || throw(DomainError(tau, "illegal value of tau"))
+    present = searchsortedfirst(precomputedquants[:, 1], tau)
+    if present <= size(precomputedquants, 1)
+        if tau == precomputedquants[present, 1]
+            return precomputedquants[present, 2] 
+        end
+    end
+
+    dnorm = Normal(0.0, 1.0)
+    if tau < 0.001 
+        return -newton(x -> tau - ChernoffComputations._cdfbar(x), ChernoffComputations._pdf, quantile(dnorm, 1.0 - tau)*0.52) 
```

Comment:
0.52 is used several times. I assume this is from the paper, in which case a variable should be used, like `xyz_const = 0.52` to convey the meaning and indicate where the reason for this value can be found

Reply:
ok I found the reference in the paper, I would suggest as follows:
Before `dnorm = Normal(0.0, 1.0)`, add a comment that one good approximation of the quantiles can be computed using `Normal(0.0, stdapprox)` with `stdapprox = 0.52`
Then use the constant `stdapprox = 0.52` below in the computations instead of the hard-coded number